examples/src/bin/custom_editable.rs

```diff
@@ -0,0 +1,140 @@
+//! # Custom GdkPaintable
+//!
+//! This example shows how to create a custom gdk::Paintable
+//! that renders a square with a linear color gradient
+
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib, graphene, gsk};
+use std::env::args;
+
+mod imp {
+    use super::*;
+
+    pub struct CustomEditable {
+        pub text: gtk::Text,
+    }
+
+    impl Default for CustomEditable {
+        fn default() -> Self {
+            Self {
+                text: gtk::Text::new(),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CustomEditable {
+        const NAME: &'static str = "CustomEditable";
+        type Type = super::CustomEditable;
+        type ParentType = gtk::Widget;
+        type Interfaces = (gtk::Editable,);
+
+        fn class_init(klass: &mut Self::Class) {
+            let total_props = klass.install_properties();
+            println!("total props: {}", total_props);
+            klass.set_layout_manager_type::<gtk::BoxLayout>();
+            klass.set_css_name("entry");
+            klass.set_accessible_role(gtk::AccessibleRole::TextBox);
+        }
+    }
+
+    impl ObjectImpl for CustomEditable {
+        fn set_property(
+            &self,
+            editable: &Self::Type,
+            id: usize,
+            value: &glib::Value,
+            pspec: &glib::ParamSpec,
+        ) {
+            println!("Setter: {} - ID: {:#?}", pspec.get_name(), id);
+            if EditableImpl::properties(self).contains(&pspec.get_name().to_string()) {
+                EditableImpl::set_property(self, editable, id, value, pspec);
+            } else {
+                unimplemented!();
+            }
+        }
+
+        fn get_property(
+            &self,
+            editable: &Self::Type,
+            id: usize,
+            pspec: &glib::ParamSpec,
+        ) -> glib::Value {
+            println!("Getter: {} - ID: {:#?}",pspec.get_name(), id);
+            if EditableImpl::properties(self).contains(&pspec.get_name().to_string()) {
+                EditableImpl::get_property(self, editable, id, pspec).unwrap()
+            } else {
+                unimplemented!();
+            }
+        }
+
+        fn constructed(&self, editable: &Self::Type) {
+            editable.init_delegate();
```

Comment:
Why is this and `finish_delegate()` needed? Can't that be automatically done somehow as part of implementing the interface?

Reply:
Not really, often, you don't want to override all the text handling functions, in such case, you would return a None for `get_delegate` and override all the other functions. But that's not what often people are after, and in such  case you have to call init_delegate/finish_delegate. Which basically just wires up/down the signal machinery from the delegate (your other Editable implementation  like gtk::Text) to your new custom Editable. So we can't really call those by default

Reply:
We could also add a bool associated constant to the `Impl` struct to automate this. Still easier than having to call a function :) Your call

Reply:
Ah, that can work, can we override the constructed/dispose functions to call them automatically somehow? if that's the case, we can remove them from the generated bindings and only call them automatically if the new associated bool is set to true.

Reply:
Hmm no, only `instance_init` currently. Not sure how the others could be covered.

Reply:
Let's worry about that in a separate PR I guess